tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,301 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute.Exceptions;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class ManagedIdentityTests
+    {
+        private static readonly string[] s_msi_scopes = { "https://management.azure.com" };
+        private static readonly string[] s_wrong_msi_scopes = { "https://managements.azure.com" };
+        private static readonly string s_clientId = "client_id";
+
+        //http proxy base URL 
+        private static readonly string s_baseURL = "https://service.msidlab.com/";
+
+        //Shared User Assigned Client ID
+        private const string UserAssignedClientID = "3b57c42c-3201-4295-ae27-d6baec5b7027";
+
+        //Non Existent User Assigned Client ID 
+        private const string NonExistentUserAssignedClientID = "3b57c42c-9999-9999-zz99-d6baec5b7027";
```

Comment:
Nit: don't use `zz` in there as it is an invalid GUID digit and a potential Guid.Parse in MSAL's logic will break your test. Consider using the Microsoft tenant id - that's a valid invalid clientid :)